From: Chad Dupuis <cdupuis@redhat.com>
Date: Fri, 3 Sep 2010 21:40:16 -0400
Subject: [scsi] qla2xxx: fix seconds_since_last_heartbeat reset
Message-id: <20100903214016.2767.92518.sendpatchset@localhost.localdomain>
Patchwork-id: 28146
O-Subject: [RHEL 5.6 PATCH 4/9] qla2xxx: Reset seconds_since_last_heartbeat
	correctly.
Bugzilla: 619814
RH-Acked-by: Rob Evers <revers@redhat.com>

Bugzilla
--------

Bug 619814 (https://bugzilla.redhat.com/show_bug.cgi?id=619814)

Upstream Status
---------------

Will be submitted upstream with our next patch submission.

Description
-----------

>From 934690f9838a5aea595cb49c84da990e9c46859f Mon Sep 17 00:00:00 2001
From: Lalit Chandivade <lalit.chandivade@qlogic.com>
Date: Thu, 8 Jul 2010 10:26:21 +0530
Subject: [PATCH] qla2xxx: Reset seconds_since_last_heartbeat correctly.

The seconds_since_last_heartbeat should be checked for consecutive
heartbeat checks. Currently it could happen that seconds_since_last_heartbeat
gets set to max (2 seconds) for non-consecutive heartbeat checks.

JIRA-Key: RHEL5.3FC-1

diff --git a/drivers/scsi/qla2xxx/qla_nx.c b/drivers/scsi/qla2xxx/qla_nx.c
index c4e2d3f..8fddfb5 100644
--- a/drivers/scsi/qla2xxx/qla_nx.c
+++ b/drivers/scsi/qla2xxx/qla_nx.c
@@ -2521,6 +2521,8 @@ qla82xx_check_fw_alive(scsi_qla_host_t *ha)
 				complete(&ha->mbx_intr_comp);
 			}
 		}
+	} else {
+		ha->seconds_since_last_heartbeat = 0;
 	}
 
 	ha->fw_heartbeat_counter = fw_heartbeat_counter;
